src/EditorFeatures/Test/CommentSelection/CSharpToggleBlockCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,315 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Editor.CSharp.CommentSelection;
+using Microsoft.CodeAnalysis.Editor.Implementation.CommentSelection;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Operations;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.CommentSelection
+{
+    [UseExportProvider]
+    public class CSharpToggleBlockCommentCommandHandlerTests : AbstractToggleBlockCommentTestBase
+    {
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleBlockComment)]
+        public void AddComment_CaretInsideToken()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        va$$r i = 1;
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+        var/**/ i = 1;
```

Comment:
interesting.  could you give some info on why this is the desirable behavior.  i'm nt sure what i think should be happening, but it seems more suitable that we would maybe comment `/*var*/`.  I don't have a strong opinion, i just want to understand the reasoning.  Thanks!

Reply:
Yes, basically we don't want to break up a valid token if they haven't selected it.  Doing this is minimally intrusive as it won't immediately cause any compilation errors.  It's hard to say what the behavior the user is expecting when they do this.
1.  They might want to add a comment about the token, in which case inserting after seems appropriate.
2.  They might want to comment out the token, in which case, your option would be appropriate.

In light of the lack of actual data, other editors seem to be having success with inserting after the identifier, so we thought we'd go with that until / unless we hear otherwise.

Reply:
> In light of the lack of actual data, other editors seem to be having success with inserting after the identifier, so we thought we'd go with that until / unless we hear otherwise.

This is a good enough reason for me.  You can close this if you want :)